tests/Unit/IO/Observers/Test_VolumeObserver.cpp

```diff
@@ -257,6 +257,74 @@ SPECTRE_TEST_CASE("Unit.IO.Observers.VolumeObserver", "[Unit][Observers]") {
+      std::add_pointer_t<element_comp>{nullptr},
+      Parallel::ArrayIndex<ElementId<2>>{ElementId<2>{element_ids[0]}});
+
+  // Although the WriteVolumeData action would typically be writing
+  // 2D surface "volume" data from a Strahlkorper, to simplify this test, here
+  // just reuse make_fake_volume_data().
+  const auto h5_write_volume_fake = make_fake_volume_data(
+      h5_write_volume_array_id, h5_write_volume_element_name + "/"s);
+
+  const std::vector<size_t> h5_write_volume_expected_extents{
+      {std::get<0>(h5_write_volume_fake)[0],
+       std::get<0>(h5_write_volume_fake)[1]}};
+  const std::vector<Spectral::Basis> h5_write_volume_expected_bases{
+      {std::get<2>(h5_write_volume_fake)[0],
+       std::get<2>(h5_write_volume_fake)[1]}};
+  const std::vector<Spectral::Quadrature> h5_write_volume_expected_quadratures{
+      {std::get<3>(h5_write_volume_fake)[0],
+       std::get<3>(h5_write_volume_fake)[1]}};
+
+  if (file_system::check_if_file_exists(h5_write_volume_file_name)) {
+    file_system::rm(h5_write_volume_file_name, true);
+  }
+
+  const observers::ObservationId write_vol_observation_id{
+      1., "ElementObservationType"};
+  runner.threaded_action<obs_writer,
+                         observers::ThreadedActions::WriteVolumeData>(
+      0, h5_write_volume_file_name, h5_write_volume_group_name,
+      write_vol_observation_id,
+      std::vector<ElementVolumeData>{{h5_write_volume_expected_extents,
+                                      std::get<1>(h5_write_volume_fake),
+                                      h5_write_volume_expected_bases,
+                                      h5_write_volume_expected_quadratures}});
+
+  {
+    std::vector<DataVector> h5_write_volume_expected_tensor_data{};
+    for (const auto& tensor_component : std::get<1>(h5_write_volume_fake)) {
+      h5_write_volume_expected_tensor_data.push_back(
+          std::get<DataVector>(tensor_component.data));
+    }
+
+    // Expected_tensor_names order is Tx, Ty, Sxx, Syy, Sxy, Syx, but
+    // h5_write_volume_tensor_data is in order Tx, Ty, Sxx, Sxy, Syx, Syy.
+    // Ensuring that the tensor data components are checked in the correct order
+    // determines the order of components in the last argument to
+    // check_volume_data.
+    TestVolumeData_detail::check_volume_data(
+        h5_write_volume_file_name + ".h5"s, 4, "element_data",
+        write_vol_observation_id.hash(), write_vol_observation_id.value(),
+        h5_write_volume_expected_tensor_data, {h5_write_volume_element_name},
+        {h5_write_volume_expected_bases},
+        {h5_write_volume_expected_quadratures},
+        {h5_write_volume_expected_extents}, expected_tensor_names,
+        {{0, 1, 2, 5, 3, 4}}, {});
+  }
+
+  if (file_system::check_if_file_exists(h5_write_volume_file_name + ".h5"s)) {
+    file_system::rm(h5_write_volume_file_name + ".h5"s, true);
+  }
+
   if (file_system::check_if_file_exists(h5_file_name)) {
```

Comment:
why not have this before the new parts you added? I'm confused on the split here

Can the new code be easily factored into a separate function so the test is easier to understand and that it's clear the test case covers 2 tests?

Reply:
Both files get written by the same process of running the mock action to interpolate (and then trigger the callbacks), so both must be checked for and removed before those actions run. It's simplest to check for these files existence at the start of the test, in my opinion. I added a comment explaining this

Reply:
O, I see. I did move the one at the end of the test, so I think we're good on that point.

I think I got confused on the ping...I didn't realize this was a separate request. Yes, I can put the surface test itself into a function, to make that clear. I'll do that in the next fixup commit

Reply:
Great! No need for a fixup, just squash right away so we don't have to wait on CI again :)